src/silx/app/view/CustomPlotSelectionWindow.py

```diff
@@ -166,14 +181,18 @@ def _addYFile(self, url=None, curve=None, node="X"):
             self.getYParent().rowCount() - 1, [iconItem, fileItem, removeItem]
         )
 
-    def _findRowContainingText(self, text):
+    def _findRowContainingText(self, text: str) -> int | None:
+        """Return the row index containing the given text, or None if not found."""
         for row in range(self.rowCount()):
             item = self.item(row, 0)
             if item and item.text() == text:
                 return row
         return None
 
-    def _createRowItems(self, filename, curve):
+    def _createRowItems(
+        self, filename: str, curve: silx.gui.plot.items.Curve
```

Comment:
`curve` might be `None` (e.g. for X)
```suggestion
        self, filename: str, curve: silx.gui.plot.items.Curve | None
```